### PR TITLE
Add pop to WoodworkTableAccessor

### LINKED ISSUE #614: Add WoodworkTableAccessor.pop
- Use pop to remove a series from a DataFrame with Woodwork initialized inplace, returning the series with Woodwork initialized
- Handle the case where Woodwork isn't initialized and return the series without woodwork initialized

### PR DESCRIPTION
**Pull request by tamargrey**
- Removes column from dataframe and returns the series with woodwork typing information initialized
- Closes #614

### CODE REVIEW

File: woodwork/table_accessor.py [hunk 1]
```diff
@@ -433,6 +433,30 @@ def _get_subset_df_with_schema(self, cols_to_include):
...
         return new_df
 
+    def pop(self, column_name):
+        """Return a Series with Woodwork typing information and remove it from the DataFrame.
+
+        Args:
+            column (str): Name of the column to pop.
+
+        Returns:
+            (pd.Series, ks.Series): Popped series with Woodwork initialized
```
Comment (thehomebrewnerd) [inline woodwork/table_accessor.py:443]: `pd.Series` only for now

File: woodwork/tests/accessor/test_table_accessor.py [hunk 2]
```diff
@@ -1397,3 +1391,65 @@ def test_set_types_errors(sample_df):
...
+    schema_df.ww.init(
+        name='table',
+        logical_types={'age': Integer},
+        semantic_tags={'age': 'custom_tag'},
+        use_standard_tags=False)
+
+    popped_series = schema_df.ww.pop('age')
+
+    assert popped_series.ww.semantic_tags == {'custom_tag'}
+    assert schema_df.ww.semantic_tags['id'] == set()
```
Comment (thehomebrewnerd) [inline woodwork/tests/accessor/test_table_accessor.py:1432]: Is this needed?

**Review by thehomebrewnerd — APPROVED** [review]
A few minor comments, but looks good!

File: woodwork/tests/accessor/test_table_accessor.py (hunk 2 above)
Comment (tamargrey) [reply woodwork/tests/accessor/test_table_accessor.py:1432]: ehhh probably not

**Pull request merged**